### PR TITLE
SO-4882: Upgrade empty extension bug

### PR DESCRIPTION
**Pull request by molnarlaura**

### CODE REVIEW

File: commons/com.b2international.index.tests/src/com/b2international/index/revision/RevisionBranchMergeTest.java [hunk 1]
```diff
@@ -586,6 +586,21 @@ public void mergeOlderTagBranchToNewerBranch() throws Exception {
...
+	@Test
+	public void mergeEmptyBranchTwice() throws Exception {
+		indexRevision(MAIN, NEW_DATA);
+		final String branchA = createBranch(MAIN, "branchA");
+		indexRevision(MAIN, NEW_DATA2);
+		final String branchB = createBranch(MAIN, "branchB");
+		branching().prepareMerge(branchA, branchB).squash(false).merge();
+		assertState(branchA, branchB, BranchState.BEHIND);
+		assertState(branchB, branchA, BranchState.FORWARD);
+        // perform another fast-forward merge to simulate sync
```
Comment (cmark) [inline commons/com.b2international.index.tests/src/com/b2international/index/revision/RevisionBranchMergeTest.java]: ⚠️ Indentation!

File: commons/com.b2international.index/src/com/b2international/index/revision/BaseRevisionBranching.java [hunk 2]
```diff
@@ -21,12 +21,17 @@
 import java.util.Collections;
 import java.util.List;
 import java.util.Map;
+import java.util.Set;
 import java.util.concurrent.Callable;
 import java.util.concurrent.TimeUnit;
 import java.util.concurrent.locks.ReentrantLock;
 import java.util.function.Consumer;
 
-import com.b2international.commons.exceptions.*;
```
Comment (cmark) [inline commons/com.b2international.index/src/com/b2international/index/revision/BaseRevisionBranching.java]: Revert this change please and configure your IDE to use wildcard imports on 5 or more import statements from the same package.

File: snomed/com.b2international.snowowl.snomed.core.rest.tests/src/com/b2international/snowowl/snomed/core/rest/ext/SnomedExtensionUpgradeTest.java [hunk 3]
```diff
@@ -19,7 +19,11 @@
 import static com.b2international.snowowl.snomed.core.rest.SnomedComponentRestRequests.getComponent;
 import static com.b2international.snowowl.snomed.core.rest.SnomedMergingRestRequests.createMerge;
 import static com.b2international.snowowl.snomed.core.rest.SnomedMergingRestRequests.waitForMergeJob;
-import static com.b2international.snowowl.snomed.core.rest.SnomedRestFixtures.*;
+import static com.b2international.snowowl.snomed.core.rest.SnomedRestFixtures.DEFAULT_LANGUAGE_CODE;
```
Comment (cmark) [inline snomed/com.b2international.snowowl.snomed.core.rest.tests/src/com/b2international/snowowl/snomed/core/rest/ext/SnomedExtensionUpgradeTest.java]: Same here.

**Review by cmark — CHANGES_REQUESTED** [review]

File: snomed/com.b2international.snowowl.snomed.core.rest.tests/src/com/b2international/snowowl/snomed/core/rest/ext/SnomedExtensionUpgradeTest.java (hunk 3 above)
Comment (molnarlaura) [reply snomed/com.b2international.snowowl.snomed.core.rest.tests/src/com/b2international/snowowl/snomed/core/rest/ext/SnomedExtensionUpgradeTest.java]: Done

File: commons/com.b2international.index/src/com/b2international/index/revision/BaseRevisionBranching.java (hunk 2 above)
Comment (molnarlaura) [reply commons/com.b2international.index/src/com/b2international/index/revision/BaseRevisionBranching.java]: Done

File: commons/com.b2international.index.tests/src/com/b2international/index/revision/RevisionBranchMergeTest.java (hunk 1 above)
Comment (molnarlaura) [reply commons/com.b2international.index.tests/src/com/b2international/index/revision/RevisionBranchMergeTest.java]: Done

**Review by cmark — APPROVED** [review]
LGTM!
